tracer/src/Datadog.Trace/Configuration/IntegrationId.cs

```diff
@@ -51,5 +51,6 @@ internal enum IntegrationId
         Process,
         HashAlgorithm,
         SymmetricAlgorithm,
+        OpenTelemetry,
```

Comment:
How should this interact with `DD_TRACE_ACTIVITY_LISTENER_ENABLED` and `DD_TRACE_OTEL_ENABLED`? 🤔 I guess we just have them all independent, but if we're checking whether the `OpenTelemetry` integration is enabled in the integrations, we should probably also check that `DD_TRACE_OTEL_ENABLED` is enabled too?

Reply:
Yeah I think that's a good idea having `DD_TRACE_OTEL_ENABLED` control whether the `OpenTelemetry` integration runs, let me put that together...

Reply:
I have `DD_TRACE_OTEL_ENABLED` and `DD_TRACE_ACTIVITY_LISTENER_ENABLED` set the same flag, in that order of precedence. And then in 5e6024b64f5cad7671d2a9256f0b60ebe69ea95c I use this config to toggle the `OpenTelemetry` integration, so everything should be off and on simultaneously

Reply:
Nice, I was thinking we were going to have to check the `DD_TRACE_OTEL_ENABLED` setting everywhere, but this is much cleaner 👍 

In which case, I'm not sure we need/should have the `IsActivityListenerEnabled` property on `ImmutableTracerSettings` any more? We should probably check whether Otel is enabled using the `IntegrationId`?

Though we may also have to think about the fact that they can toggle the `IntegrationId` _after_ the app has started&mdash;currently we check `IsActivityListenerEnabled` in the initial load in `Instrumentation`. _Technically_ they could disable/enable it in code later, and we wouldn't update accordingly

Reply:
This will be...challenging. Maybe we can solve this later 😅